src/pkg/exporter/project_collector.go

```diff
@@ -0,0 +1,239 @@
+package exporter
+
+import (
+	"encoding/json"
+	"errors"
+
+	"github.com/goharbor/harbor/src/common/dao"
+	"github.com/goharbor/harbor/src/lib/log"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// ProjectCollectorName ...
+const ProjectCollectorName = "ProjectCollector"
+
+var (
```

Comment:
Please make sure the SQLs are tested in the integration test otherwise a change in DB schema will break it.